### DISCUSSION TITLE
增加 MYCFLAGS 参数时，编译出错

### DISCUSSION THREAD

**Discussion by samuelyao314** [cloudwu/skynet#1161] [Ideas]
操作系统：centos 7， 增加 MYCFLAGS 参数时，编译出错

$ make linux MYCFLAGS=-g
make all PLAT=linux SKYNET_LIBS="-lpthread -lm -ldl -lrt" SHARED="-fPIC --shared" EXPORT="-Wl,-E" MALLOC_STATICLIB="3rd/jemalloc/lib/libjemalloc_
pic.a" SKYNET_DEFINES=""
make[1]: 进入目录“/home/samuelyao/terminator/skynet”
cd 3rd/lua && make CC='cc -std=gnu99' linux
make[2]: 进入目录“/home/samuelyao/terminator/skynet/3rd/lua”
make all SYSCFLAGS="-DLUA_USE_LINUX" SYSLIBS="-Wl,-E -ldl -lreadline"
make[3]: 进入目录“/home/samuelyao/terminator/skynet/3rd/lua”
cc -std=gnu99 -O2 -Wall -Wextra -DLUA_USE_LINUX -g   -c -o lapi.o lapi.c
cc -std=gnu99 -O2 -Wall -Wextra -DLUA_USE_LINUX -g   -c -o lcode.o lcode.c
cc -std=gnu99 -O2 -Wall -Wextra -DLUA_USE_LINUX -g   -c -o lctype.o lctype.c
cc -std=gnu99 -O2 -Wall -Wextra -DLUA_USE_LINUX -g   -c -o ldebug.o ldebug.c
cc -std=gnu99 -O2 -Wall -Wextra -DLUA_USE_LINUX -g   -c -o ldo.o ldo.c
cc -std=gnu99 -O2 -Wall -Wextra -DLUA_USE_LINUX -g   -c -o ldump.o ldump.c
cc -std=gnu99 -O2 -Wall -Wextra -DLUA_USE_LINUX -g   -c -o lfunc.o lfunc.c
cc -std=gnu99 -O2 -Wall -Wextra -DLUA_USE_LINUX -g   -c -o lgc.o lgc.c
cc -std=gnu99 -O2 -Wall -Wextra -DLUA_USE_LINUX -g   -c -o llex.o llex.c
cc -std=gnu99 -O2 -Wall -Wextra -DLUA_USE_LINUX -g   -c -o lmem.o lmem.c
cc -std=gnu99 -O2 -Wall -Wextra -DLUA_USE_LINUX -g   -c -o lobject.o lobject.c
cc -std=gnu99 -O2 -Wall -Wextra -DLUA_USE_LINUX -g   -c -o lopcodes.o lopcodes.c
cc -std=gnu99 -O2 -Wall -Wextra -DLUA_USE_LINUX -g   -c -o lparser.o lparser.c
cc -std=gnu99 -O2 -Wall -Wextra -DLUA_USE_LINUX -g   -c -o lstate.o lstate.c
cc -std=gnu99 -O2 -Wall -Wextra -DLUA_USE_LINUX -g   -c -o lstring.o lstring.c
lstring.c:24:20: 致命错误：atomic.h：没有那个文件或目录
 #include "atomic.h"

**Comment by cloudwu** [comment]
目前 skynet 的 Makefile 不支持 MYCFLAGS ，即使写了，目前的编译命令大多也没有加上它。你无法用 MYCFLAGS=-g 加上 -g 编译选项。

**Comment by samuelyao314** [comment]
请问，未来有支持   -g  的计划吗？
